Math/836RectangleOverlap.py

```
@@ -0,0 +1,47 @@
+Constraints:
+
+rec1.length == 4
+rec2.length == 4
+-109 <= rec1[i], rec2[i] <= 109
+rec1 and rec2 represent a valid rectangle with a non-zero area.
+
+
+'''
+class Solution:
+    def isRectangleOverlap(self, rec1: list[int], rec2: list[int]) -> bool:
+          if rec1[0]>=rec2[2] or rec2[0]>=rec1[2]: return False 
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)